fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java

```diff
@@ -38,23 +34,35 @@ public class CloudConfigurationFactory {
             new AzureCloudConfigurationProvider(),
             new GCPCloudConfigurationProvoder(),
             new AliyunCloudConfigurationProvider(),
-            new HDFSCloudConfigurationProvider());
+            new HDFSCloudConfigurationProvider(),
+            new CloudConfigurationProvider() {
+                @Override
+                public CloudConfiguration build(Map<String, String> properties) {
+                    return new CloudConfiguration();
+                }
+            });
 
     public static CloudConfiguration buildCloudConfigurationForStorage(Map<String, String> properties) {
         for (CloudConfigurationProvider factory : cloudConfigurationFactoryChain) {
             CloudConfiguration cloudConfiguration = factory.build(properties);
             if (cloudConfiguration != null) {
+                cloudConfiguration.loadCommonFields(properties);
                 return cloudConfiguration;
             }
         }
-
-        return buildDefaultCloudConfiguration();
+        // Should never reach here.
+        return null;
     }
 
     public static AWSCloudCredential buildGlueCloudCredential(HiveConf hiveConf) {
-        AWSCloudConfigurationProvider awsCloudConfigurationProvider =
-                (AWSCloudConfigurationProvider) cloudConfigurationFactoryChain.get(0);
-        return awsCloudConfigurationProvider.buildGlueCloudCredential(hiveConf);
+        for (CloudConfigurationProvider factory : cloudConfigurationFactoryChain) {
+            if (factory instanceof AWSCloudConfigurationProvider) {
+                AWSCloudConfigurationProvider provider = ((AWSCloudConfigurationProvider) factory);
+                return provider.buildGlueCloudCredential(hiveConf);
+            }
+        }
+        // Should never reach here.
```

Comment:
why not reach here, if user enter invalid glue parameters, it will build glue cloud credential failed.